### PR TITLE
feat(product-service): init

### PR DESCRIPTION
**Pull request by web-arch**

### CODE REVIEW

File: product-service/handler.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { ProductService } from './src/services/productService';
+
+import { getProductByIdHandler } from './src/getProductById';
+import { getAllProductsHandler } from './src/getAllProducts';
+
+const productService = new ProductService();
```
Comment (VarvaraSemichevaEpam) [inline product-service/handler.ts:6]: [minor]

I assume that the reason that you move this service initialization this way and pass it as the parameter - you are trying to organize something like dependency inversion. Basically it makes sense from testing point of view, of course.
But it's quite a high level of abstraction and since JS is dynamically typed and can be easily mocked - it's not so essential to provide such mechanisms.
To recap - it's completely correct to use it, but you can also require and initialize services in direct handlers and it's also be valid.

File: product-service/src/utils/apiResponseHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const defaultHeaders = {
+    'Access-Control-Allow-Methods': '*',
+    'Access-Control-Allow-Headers': '*',
+    'Access-Control-Allow-Origin': '*'
+};
+
+export function getSuccessResponse(
+    {
+        data,
+        statusCode = 200
```
Comment (VarvaraSemichevaEpam) [inline product-service/src/utils/apiResponseHelpers.ts:12]: [minor]
Better to use enum for HTTP codes for readability

File: product-service/src/utils/winstonLogger.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        this.logger = createLogger({
+            level: process.env.ENV_STAGE === 'prod' ? 'error' : 'info',
+            transports: [
+                new transports.Console({
+                    format: this.format
+                })
+            ]
+        });
+    }
+    logRequest( message: string ){
```
Comment (VarvaraSemichevaEpam) [inline product-service/src/utils/winstonLogger.ts:29]: This function is never used

**Review by VarvaraSemichevaEpam — COMMENTED** [review]
Just some general concerns for the **future** evaluation of this implementation:

1) Good, that you've already introduced a logger - do not forget to always log request and response
2) Request validation
3) Mappings - we shouldn't store any extra data in DB and we should not return any extra data to client in advance - introducing DT objects
4) Structure - I think it's already pretty good since you introduced services layer.
5) Debugging - how did you check the lambda? You can take a look at the 'serverless-offline' package.
